core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -750,20 +750,27 @@ class ProducerStateManager(val topicPartition: TopicPartition,
     lastMapOffset = 0L
   }
 
+  def firstUnstableOffsetAfter(completedTxn: CompletedTxn): Long = {
+    val ongoingTxnsIter = ongoingTxns.values.iterator()
+    while (ongoingTxnsIter.hasNext) {
+      val ongoingTxn = ongoingTxnsIter.next()
+      if (ongoingTxn.producerId != completedTxn.producerId)
+        return ongoingTxn.firstOffset.messageOffset
+    }
+    completedTxn.lastOffset + 1
```

Comment:
Did you intentionally use a `while` loop here instead of `find`?

Reply:
Just got a little lazy since I didn't have the magic converter api already imported.